FAILED: CachingDirectivesExamplesTest.testCreateCache

https://jenkins.akka.io:8498/job/pr-validator-akka-http/1757/

Previous work on this test: #1541 

```
Error Message

Unexpected entity! expected:<...ample.com/1 @ count [500]> but was:<...ample.com/1 @ count [1]>   Request was:      HttpRequest(HttpMethod(GET),/1,List(),HttpEntity.Strict(none/none,ByteString()),HttpProtocol(HTTP/1.1))   Route result was: FulfilledFuture(Complete(HttpResponse(200 OK,List(),HttpEntity.Strict(text/plain; charset=UTF-8,Request for http://example.com/1 @ count 1),HttpProtocol(HTTP/1.1)))) 
Stacktrace

java.lang.AssertionError: Unexpected entity! expected:<...ample.com/1 @ count [500]> but was:<...ample.com/1 @ count [1]>
  Request was:      HttpRequest(HttpMethod(GET),/1,List(),HttpEntity.Strict(none/none,ByteString()),HttpProtocol(HTTP/1.1))
  Route result was: FulfilledFuture(Complete(HttpResponse(200 OK,List(),HttpEntity.Strict(text/plain; charset=UTF-8,Request for http://example.com/1 @ count 1),HttpProtocol(HTTP/1.1))))

	at akka.http.javadsl.testkit.JUnitRouteTestBase$$anon$1.reportDetails(JUnitRouteTest.scala:45)
	at akka.http.javadsl.testkit.JUnitRouteTestBase$$anon$1.assertEquals(JUnitRouteTest.scala:30)
	at akka.http.javadsl.testkit.TestRouteResult.assertEqualsKind(TestRouteResult.scala:227)
	at akka.http.javadsl.testkit.TestRouteResult.assertEntity(TestRouteResult.scala:166)
	at docs.http.javadsl.server.directives.CachingDirectivesExamplesTest.testCreateCache(CachingDirectivesExamplesTest.java:176)
```

--- a/docs/src/test/java/docs/http/javadsl/server/directives/CachingDirectivesExamplesTest.java
+++ b/docs/src/test/java/docs/http/javadsl/server/directives/CachingDirectivesExamplesTest.java
@@ -160,19 +160,7 @@ public class CachingDirectivesExamplesTest extends JUnitRouteTest {
     final Route route = cache(lfuCache, keyerFunction, () -> innerRoute);
     //#create-cache
 
-    // tests:
-    testRoute(route)
-      .run(HttpRequest.GET("/1"))
-      .assertEntity("Request for http://example.com/1 @ count 1");
-
-    for (int i = 1; i < 500; i++) {
-      testRoute(route)
-        .run(HttpRequest.GET("/" + i))
-        .assertEntity("Request for http://example.com/" + i + " @ count " + i);
-    }
-
-    testRoute(route)
-      .run(HttpRequest.GET("/1"))
-      .assertEntity("Request for http://example.com/1 @ count 500");
+    // We don't test the eviction settings here. Deterministic testing of eviction is hard because
+    // caffeine's LFU is probabilistic.
   }
 }

--- a/docs/src/test/scala/docs/http/scaladsl/server/directives/CachingDirectivesExamplesSpec.scala
+++ b/docs/src/test/scala/docs/http/scaladsl/server/directives/CachingDirectivesExamplesSpec.scala
@@ -126,19 +126,7 @@ class CachingDirectivesExamplesSpec extends RoutingSpec with CachingDirectives {
     val route = cache(lfuCache, keyerFunction)(innerRoute)
     //#create-cache
 
-    // tests:
-    Get("/1") ~> route ~> check {
-      responseAs[String] shouldEqual "Request for http://example.com/1 @ count 1"
-    }
-
-    for (i <- 1 until 500) {
-      Get(s"/$i") ~> route ~> check {
-        responseAs[String] shouldEqual s"Request for http://example.com/$i @ count $i"
-      }
-    }
-
-    Get("/1") ~> route ~> check {
-      responseAs[String] shouldEqual "Request for http://example.com/1 @ count 500"
-    }
+    // We don't test the eviction settings here. Deterministic testing of eviction is hard because
+    // caffeine's LFU is probabilistic.
   }
 }

